toolchain/lex/tokenized_buffer.cpp

```diff
@@ -291,20 +298,20 @@ auto TokenizedBuffer::PrintToken(llvm::raw_ostream& output_stream,
                     << "`";
       break;
     default:
-      if (token_info.kind.is_opening_symbol()) {
+      if (token_info.kind().is_opening_symbol()) {
         output_stream << ", closing_token: "
                       << GetMatchedClosingToken(token).index;
-      } else if (token_info.kind.is_closing_symbol()) {
+      } else if (token_info.kind().is_closing_symbol()) {
         output_stream << ", opening_token: "
                       << GetMatchedOpeningToken(token).index;
       }
       break;
   }
 
-  if (token_info.has_trailing_space) {
-    output_stream << ", has_trailing_space: true";
+  if (token_info.has_leading_space()) {
+    output_stream << ", has_leading_space: true";
   }
-  if (token_info.is_recovery) {
+  if (!recovery_tokens_.empty() && recovery_tokens_[token_index]) {
```

Comment:
Can this use the helper?

```suggestion
  if (IsRecoveryToken(token)) {
```